modules/extensions/galasa-extensions-parent/dev.galasa.extensions.common/src/main/java/dev/galasa/extensions/common/couchdb/CouchdbStore.java

```diff
@@ -147,13 +148,43 @@ protected List<ViewRow> getAllDocsFromDatabase(String dbName) throws CouchdbExce
      * @throws CouchdbException if there was a problem accessing the CouchDB store
      *                          or its response
      */
-    protected <T> T getDocumentFromDatabase(String dbName, String documentId, Class<T> classOfObject)
+    public <T> T getDocumentFromDatabase(String dbName, String documentId, Class<T> classOfObject)
             throws CouchdbException {
         HttpGet getDocumentRequest = httpRequestFactory.getHttpGetRequest(storeUri + "/" + dbName + "/" + documentId);
 
         return gson.fromJson(sendHttpRequest(getDocumentRequest, HttpStatus.SC_OK), classOfObject);
     }
 
+    public ViewResponse getDocumentsFromDatabaseViewByKey(String dbName, String viewName, String queryKey, boolean includeDocs)
+            throws CouchdbException {
+        ViewResponse viewResponse = null;
+        String viewRequestUriStr = storeUri + "/" + dbName + "/_design/docs/_view/" + viewName;
+        try {
+            URIBuilder uriBuilder = new URIBuilder(viewRequestUriStr);
+
+            // CouchDB requires the value for the "key" parameter to be quoted
+            uriBuilder.addParameter("key", '"' + queryKey + '"');
+
+            if (includeDocs) {
+                uriBuilder.addParameter("include_docs", "true");
+            }
+
+            HttpGet getRunsRequest = httpRequestFactory.getHttpGetRequest(uriBuilder.build().toString());
+
+            String responseEntity = sendHttpRequest(getRunsRequest, HttpStatus.SC_OK);
+            viewResponse = gson.fromJson(responseEntity, ViewResponse.class);
+
+            if (viewResponse.rows == null) {
+                String errorMessage = ERROR_FAILED_TO_GET_VIEW_DOCUMENTS_FROM_DATABASE.getMessage(viewName, dbName);
+                throw new CouchdbException(errorMessage);
+            }
+        } catch (URISyntaxException e) {
+            String errorMessage = ERROR_URI_IS_INVALID.getMessage(viewRequestUriStr);
+            throw new CouchdbException(errorMessage, e);
+        }
+        return viewResponse;
+    }
```

Comment:
Wondering if the couchdb part of the extensions/common should be in a separate ...common.couchdb bundle. As the etcd extensions don't really need this code ?

Reply:
Thanks, added `dev.galasa.extensions.common.couchdb` bundle in https://github.com/galasa-dev/galasa/pull/108/commits/adc238a09a558157c79900303f414bc140af090a